app/pages/confirm.jsx

```diff
@@ -17,15 +18,59 @@ class Confirm extends React.Component {
     constructor() {
         super();
 
+        this.state = {
+            notifications: OrderedSet()
+        };
+
         this.onSubmit = this.onSubmit.bind(this);
     }
 
+    addNotification(message, key, action) {
+        return this.setState({
+            notifications: this.state.notifications.add({
+                message,
+                key,
+                action: action || 'Dismiss',
+                onClick: (notification, deactivate) => {
+                    deactivate();
+                    this.removeNotification(key);
+                },
+                dismissAfter: 5000
+            })
+        });
+    }
+
+    removeNotification(key) {
+        this.setState({
+            notifications: this.state.notifications.filter(n => n.key !== key)
+        });
+    }
+
     componentDidMount() {
         this.props.dispatch(ConfirmationThunks.loadConfirmation());
+        this.props.dispatch(ConfirmationThunks.loadForm());
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (
+            nextProps.userState.data.form &&
+            nextProps.userState.data.confirmation
+        ) {
+            for (var i in nextProps.userState.data.confirmation) {
+                if (i in nextProps.userState.data.form) {
+                    nextProps.userState.data.form[i].default =
```

Comment:
This is setting the values of the form when it was previously saved and is newly loaded (for example, the confirmation page would load the saved confirmation stuff, but it wouldn't be loaded into the form)